src/Pact/Types/Util.hs

```diff
@@ -142,10 +145,44 @@ outputJSON a = BSL8.putStrLn $ encode a
 data RenderColor = RColor | RPlain
 
 renderString :: Pretty a => (Doc -> SimpleDoc) -> RenderColor -> a -> String
-renderString renderf colors p = displayS (renderf ((case colors of RColor -> id; RPlain -> plain) (pretty p))) ""
+renderString renderf colors p =
+  let display = case colors of
+        RColor -> displayS . renderf         . pretty
+        RPlain -> displayS . renderf . plain . pretty
+  in display p ""
 
 renderCompactString :: Pretty a => a -> String
-renderCompactString = renderString renderCompact RPlain
+renderCompactString = renderString renderReallyCompact RPlain
+
+renderCompactText :: Pretty a => a -> Text
+renderCompactText = pack . renderCompactString
+
+-- This is the same as @renderCompact@ from the pretty printing package, except
+-- that printer inserts line breaks every chance it gets whereas this function
+-- never inserts them.
+renderReallyCompact :: Doc -> SimpleDoc
```

Comment:
Are we happy with `ansi-wl-pprint` then? If we're making this big change we might as well be sure we like it, esp. if we're going to maintain this monster :)

Reply:
I'd rather use `prettyprinter`: #367

Reply:
Do you want to do that now then?

Reply:
done